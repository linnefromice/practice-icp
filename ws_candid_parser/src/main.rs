use std::path::Path;

use anyhow::Result;
use candid::pretty_check_file;

fn main() -> Result<()> {
    let candid_path = "assets/sample.did";
    let (env, actor) = pretty_check_file(Path::new(candid_path))?;
    let config = candid::bindings::rust::Config::new();

    let result = candid::bindings::rust::compile(&config, &env, &None);
    println!("{}", result);

    if let Some(actor) = actor {
        let result = candid::bindings::rust::compile(&config, &env, &Some(actor));
        println!("{}", result);
    }

    Ok(())
}
