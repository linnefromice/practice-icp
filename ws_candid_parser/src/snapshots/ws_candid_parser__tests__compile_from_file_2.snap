---
source: src/main.rs
expression: result
---
// This is an experimental feature to generate Rust binding from Candid.
// You may want to manually adjust some of the types.
#![allow(dead_code, unused_imports)]
use candid::{self, CandidType, Deserialize, Principal, Encode, Decode};
use ic_cdk::api::call::CallResult as Result;

#[derive(CandidType, Deserialize)]
pub struct CalculateArgs { pub target: String, pub index: u64 }

#[derive(CandidType, Deserialize)]
pub struct CanisterMetricsSnapshot {
  pub cycles: candid::Nat,
  pub timestamp: u64,
}

#[derive(CandidType, Deserialize)]
pub enum Env { Production, Test, LocalDevelopment }

#[derive(CandidType, Deserialize)]
pub enum InitError {
  InvalidDestination(String),
  InvalidPrincipal(Principal),
  InvalidContent(String),
  InvalidRequest(String),
}

#[derive(CandidType, Deserialize)]
pub struct LensValue { pub dummy: u64 }

#[derive(CandidType, Deserialize)]
pub enum Result_ { Ok, Err(InitError) }

#[derive(CandidType, Deserialize)]
pub enum SourceType {
  #[serde(rename="evm")]
  Evm,
  #[serde(rename="https")]
  Https,
  #[serde(rename="chainsight")]
  Chainsight,
}

#[derive(CandidType, Deserialize)]
pub struct Sources {
  pub source: String,
  pub interval_sec: Option<u32>,
  pub attributes: Vec<(String,String,)>,
  pub source_type: SourceType,
}


